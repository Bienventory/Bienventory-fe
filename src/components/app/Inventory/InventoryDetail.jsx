import { Modal, Typography } from '@mui/material';
import React from 'react';
import PlusButton from '../../Buttons/PlusButton';
import MinusButton from '../../Buttons/MinusButton';
import { Box } from '@mui/system';
import Button from '@mui/material/Button';
import ButtonGroup from '@mui/material/ButtonGroup';
import { Link } from 'react-router-dom';
import InventoryForm from './InventoryForm';
// import { useParams } from 'react-router-dom';

const style = {
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: 400,
  bgcolor: 'white',
  border: '2px solid #000',
  boxShadow: 24,
  p: 4,
};

export default function InventoryDetail() {
  // const { id } = useParams();
  // const inventoryDetail = getById(id);

  const inventoryDetail = {
    id: 1,
    item_name: 'potatoes',
    description: 'starch, staple',
    total_on_hand: 10,
    unit_type: 'lbs',
    restock_level: 2,
  };

  const {
    id,
    item_name,
    description,
    total_on_hand,
    unit_type,
    restock_level,
  } = inventoryDetail;

  const [open, setOpen] = React.useState(false);
  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);

  return (
    <div>
      <Link
        to={
          ({ pathname: `/inventory/${id}` },
          { releasesProps: { item: inventoryDetail } })
        }
      >
        <ButtonGroup
          variant="contained"
          aria-label="outlined primary button group"
        >
          <Button onClick={handleOpen}>Edit</Button>
          <Modal
            open={open}
            onClose={handleClose}
            aria-labelledby="modal-modal-title"
            aria-describedby="modal-modal-description"
          >
            <Box sx={style}>
              <InventoryForm />
            </Box>
          </Modal>
          <Button>Delete</Button>
        </ButtonGroup>
      </Link>

      <Typography variant="h2">{item_name}</Typography>
      <Typography variant="h6">{description}</Typography>
      <Typography variant="h6">
        Restock at {restock_level} {unit_type}
      </Typography>
      <Typography variant="h2" align="center">
        {total_on_hand} {unit_type}
      </Typography>
      <Box sx={{ display: 'flex', justifyContent: 'space-around' }}>
        <MinusButton />
        <PlusButton />
      </Box>
    </div>
  );
}
