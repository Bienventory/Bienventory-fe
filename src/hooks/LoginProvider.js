import React, { createContext, useContext, useState } from 'react';
import PropTypes from 'prop-types';

export const Context = createContext();

//context: user authentication, inventory array, menu array

const LoginProvider = ({ children }) => {
  const [user, setUser] = useState('');
  const [loggedIn, setloggedIn] = useState(false);

  const logInUser = async (googleId) => {
    // const response = await fetch(`theapi.urrl.com///${googleId}`);
    // const userObj = await response.json();
    const userObj = { google_id: googleId, notifications: false };

    setUser(userObj);
    setloggedIn(true);
  };

  const logOutUser = () => {
    setUser('');
    setloggedIn(false);
  };

  return (
    <Context.Provider value={{ user, loggedIn, logInUser, logOutUser }}>
      {children}
    </Context.Provider>
  );
};

export const uselogInUser = () => {
  const { logInUser } = useContext(Context);
  return logInUser;
};

export const uselogOutUser = () => {
  const { logOutUser } = useContext(Context);
  return logOutUser;
};

export const useloggedIn = () => {
  const { loggedIn } = useContext(Context);
  return loggedIn;
};

export const useUser = () => {
  const { user } = useContext(Context);
  return user;
};

LoginProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default LoginProvider;
