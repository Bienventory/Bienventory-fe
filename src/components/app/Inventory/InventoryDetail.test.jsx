import React from 'react';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import InventoryDetail from './InventoryDetail';
import inventoryDetail from '../../../fixtures/inventoryDetail.json';
import LoginProvider from '../../../hooks/LoginProvider';

const server = setupServer(
  rest.get(
    'https://bienventory.herokuapp.com/api/v1/inventory/:id',
    (req, res, ctx) => {
      return res(ctx.json(inventoryDetail));
    }
  )
);

describe('Inventory Detail component', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays the inventory on the dashboard', async () => {
    const { container } = render(
      <LoginProvider>
        <MemoryRouter>
          <InventoryDetail />
        </MemoryRouter>
      </LoginProvider>
    );
    
    const h2 = await screen.findByText('butter');
    screen.getByText('Edit');
    screen.getByText('Delete');
    

    expect(h2).not.toBeEmptyDOMElement;
    expect(container).toMatchSnapshot();
  });
});
