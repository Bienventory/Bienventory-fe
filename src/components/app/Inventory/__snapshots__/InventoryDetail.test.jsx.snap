// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory Detail component displays the inventory on the dashboard 1`] = `
<div>
  <div>
    <div
      aria-label="outlined primary button group"
      class="MuiButtonGroup-root MuiButtonGroup-contained css-dr3dyz-MuiButtonGroup-root"
      role="group"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Edit
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Delete
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <h2
      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
    >
      butter
    </h2>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      unsalted butter
    </h6>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Restock at 
      5
       
      lbs
    </h6>
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
    >
      24.099999999999998
       
      lbs
    </h2>
  </div>
</div>
`;
